src/arm7tdmi.rs

```diff
@@ -132,4 +175,25 @@ mod tests {
         let output = ArmModeInstruction::get_instruction(0b1110_1010_0000_0000_0000_0000_0111_1111);
         assert_eq!(output, Ok(ArmModeInstruction::Branch));
     }
+
+    #[test]
+    fn check_mov_rx_immediate() {
+        // MOV R0, 0
+        let mut opcode: u32 = 0b11100011101000000000000000000000;
+
+        // MOV Rx,x
+        let mut cpu = Arm7tdmi::new(vec![]);
+        for rx in 0..16u32 {
+            let change_op = rx << 12;
+            opcode = (opcode & 0xFF_FF_0F_FF) + change_op;
+            opcode = (opcode & 0xFF_FF_FF_00) + rx;
```

Comment:
I understand this test more or less but is not super clear the intention, there magic number like `0xFF_FF_0F_FF` and `0xFF_FF_FF_00` and operation with `rx` that goes `0..15` but the test is solid

Reply:
The intention is to modify the opcode resulting a cycle like this:
```
0 -> MOV R0, 0,
1 -> MOV R1, 1
...
15 -> MOV R15, 15
```
I'll add some comments above the magic numbers

Reply:
Yes I got it, but that is the internal opcode, not "the main", is the opcode that regards only `MOV`, sure is right but coming from other context more clear code avoid cognitive load for understand this every time :D